### PR TITLE
vault: retry reading on errors

### PR DESCRIPTION
**Pull request by DeviaVir**
any data source might have temporary errors, let's at least give this provider a chance to retry before returning the error

the timeout is customizable on the `provider {}` block

### CODE REVIEW

File: customconfig/config.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package customconfig
 
 type Config struct {
+	TimeoutMinutes int
```
Comment (mathyourlife-fitbit) [inline customconfig/config.go]: A `TimeoutMinutes` would prevent being able to set fractional minute timeouts such as 30 seconds both here because it's an `int` and when used to cast to a `time.Duration` as `time.Duration(0.5) * time.Minute` throws `constant 0.5 truncated to integer`.  Can we use `TimeoutSeconds` instead?

File: customconfig/data_source_vault_token.go [hunk 2]
```diff
@@ -98,38 +100,57 @@ func vaultTokenDataSourceRead(d *schema.ResourceData, meta interface{}) error {
...
 	}
 
 	data := string(body)
 	if debug {
 		log.Printf("[DEBUG] Got %s from Vault", string(data))
 	}
-	d.Set("data_json", string(data))
 
-	var secret map[string]interface{}
 	json.Unmarshal([]byte(data), &secret)
```
Comment (mathyourlife-fitbit) [inline customconfig/data_source_vault_token.go]: Realize it's not part of this change, but could use `json.Unmarshal(body, &secret)` here instead of recasting the data.

File: customconfig/utils.go [hunk 3]
```diff
@@ -3,9 +3,55 @@ package customconfig
...
 	return hex.EncodeToString(sha[:])
 }
+
+func retry(retryFunc func() error, minutes int) error {
+	return retryTime(retryFunc, minutes)
+}
+
+func retryTime(retryFunc func() error, minutes int) error {
+	wait := 1
+	return resource.Retry(time.Duration(minutes)*time.Minute, func() *resource.RetryError {
```
Comment (mathyourlife-fitbit) [inline customconfig/utils.go]: fwiw, `Deprecated: Please use RetryContext to ensure proper plugin shutdown`
https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/resource#Retry

Comment (DeviaVir) [reply customconfig/utils.go]: Using latest 1.13.0: `customconfig/utils.go:28:9: undefined: resource.RetryContext`

Is this coming from v2? (which only has a RC https://github.com/hashicorp/terraform-plugin-sdk/releases)
